### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'SOME_VARIABLE')

### DISCUSSION THREAD

**Discussion by rostber** [webpack/webpack#16573] [Q&A]
# Bug report

**What is the current behavior?**
An error sometimes happens with an import path. It depends on a reexport variable with another code influence. The error looks like:
```
TypeError
Cannot read properties of undefined (reading 'FETCH_STOREFRONT_SECTIONS')
```

The same issue https://github.com/apollographql/apollo-client/issues/9659 doesn't solve the problem by solution:
```
"@apollo/client": "3.6.2",
"graphql": "15.8.0",
```

**If the current behavior is a bug, please provide the steps to reproduce.**
I don't know exactly way to reproduce it. Some reexport combinations do it or not. It disappears if the import path changes. It's really unpredictable error in the app code.

**With error** (It is reexport through index.ts: `export { FETCH_STOREFRONT_SECTIONS } from './fetchStorefrontSections.gql'`):

```js
import { FETCH_STOREFRONT_SECTIONS } from 'src/entities/Storefront/api/fetchStorefrontSections'
```

**Without error** (It is direct path):

```js
import { FETCH_STOREFRONT_SECTIONS } from 'src/entities/Storefront/api/fetchStorefrontSections/fetchStorefrontSections.gql'
```

I found out, that error throws from generated bundle.js code:

```js
......
/***/ "./src/entities/Storefront/api/fetchStorefrontSections/index.ts":
  /*!***********************************************************************!*\
    !*** ./src/entities/Storefront/api/fetchStorefrontSections/index.ts ***!
    \***********************************************************************/
  /***/ (function(module, __webpack_exports__, __webpack_require__) {
  
  "use strict";
  __webpack_require__.r(__webpack_exports__);
  /* harmony export */ __webpack_require__.d(__webpack_exports__, {
  /* harmony export */   "FETCH_STOREFRONT_SECTIONS": function() { return /* reexport safe */ _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__.FETCH_STOREFRONT_SECTIONS; },
  /* harmony export */ });
  /* harmony import */ var _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./fetchStorefrontSections.gql */ "./src/entities/Storefront/api/fetchStorefrontSections/fetchStorefrontSections.gql.ts");
  __webpack_require__.$Refresh$.runtime = __webpack_require__(/*! ./node_modules/react-refresh/runtime.js */ "./node_modules/react-refresh/runtime.js");
......
```

If the `_fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__` variable declaration and the assignment `FETCH_STOREFRONT_SECTIONS` property was swapped, the error would be disappear:

```js
var _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__("./src/entities/Storefront/api/fetchStorefrontSections/fetchStorefrontSections.gql.ts");

__webpack_require__.d(__webpack_exports__, {
  "FETCH_STOREFRONT_SECTIONS": function() { return _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__.FETCH_STOREFRONT_SECTIONS; },
});
```

**What is the expected behavior?**
Anyone of combination reexport paths shouldn't throw error in the runtime.

**Other relevant information:**
webpack version:  5.75.0 or 5.73.0.
Node.js version: 18.12.0 or 16.12.0
Operating System: osx or linux
Additional tools: graphql-tag

**Comment by alexander-akait** [comment]
What is webpack configuration? Looks like you have something strange in `resolver`, check you don't have `import { FETCH_STOREFRONT_SECTIONS } from 'src/entities/Storefront/api/fetchStorefrontSections/fetchStorefrontSections.(js|jsx|ts|tsx)'` files, webpack can try to resolve them firstly, anyway I strongly recommend to extension at the end of import

**Reply by rostber** [reply]
The resolver of webpack configuration looks like:
```
resolve: {
  alias: {
    src: '/Users/some-user/some-project/src',
    react: '/Users/some-user/some-project/node_modules/react',
    'react-router-dom': '/Users/some-user/some-project/node_modules/react-router-dom',
    'react-i18next': '/Users/some-user/some-project/node_modules/react-i18next'
  },
  extensions: [ '.ts', '.tsx', '.js' ],
  symlinks: false,
}
```

PS: Import paths without extension it is typical thing. In my bundle.js example you can see, that file path was generated correct.

**Reply by alexander-akait** [reply]
I am afraid for `src`, `src` is often used name for library directories, I recommend to use `@src` locally (or any special character), can you try to change it?

**Reply by rostber** [reply]
It doesn't solve the problem.

The error disappears if `bundle.js` gets new order of rows:
```js
var _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__("./src/entities/Storefront/api/fetchStorefrontSections/fetchStorefrontSections.gql.ts");

__webpack_require__.d(__webpack_exports__, {
  "FETCH_STOREFRONT_SECTIONS": function() { return _fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__.FETCH_STOREFRONT_SECTIONS; },
});
```
It gets a guarantee, that `_fetchStorefrontSections_gql__WEBPACK_IMPORTED_MODULE_0__` is going to be exist. Why is the order different?

**Reply by alexander-akait** [reply]
@rostber Maybe you can create small example like you use in your projects, I still think something wrong with alias, also check your `tsconfig.js`, maybe your packages marked as commonjs, but use ESM (or vice versa)
